apps/src/sites/studio/pages/signup.js

```diff
@@ -115,18 +123,96 @@ window.SignupManager = function (options) {
 
     // Force teachers to explicitly accept terms of service.
     $("#user_terms_of_service_version").prop('checked', false);
+
+    logTeacherToggle(true);
+    lastUserType = "teacher";
+  }
+
+  /**
+   * Log signup-related analytics events to Firehose
+   * @param eventName name of the event to log
+   * @param extraData optional hash object for supplemental data (will show up in the data_json field)
+   */
+  function logAnalyticsEvent(eventName, extraData = {}) {
+    if (!self.uuid) {
+      if (!!window.localStorage && !!window.localStorage.getItem("analyticsID")) {
```

Comment:
The idea here is to make the uuid somewhat persistent, though it won't follow logged in users across different machines/browsers, which means it won't be very useful for many students beyond a single session. They'll typically have CDO user IDs, though, which we can use for other tracking. For the purposes of signup page instrumentation, all that really matters is that it's the same during a single attempt at account creation, so this should suffice for now.